File: starterAIs/SS_Starter.ts
interface Vector {
  x: number
  y: number
}

interface FishDetail {
  color: number
  type: number
}

interface Fish {
  fishId: number
  pos: Vector
  speed: Vector
  detail: FishDetail
}

interface Drone {
  droneId: number
  pos: Vector
  dead: number
  battery: number
  scans: number[]
}

interface RadarBlip {
  fishId: number
  dir: string
}


const fishDetails = new Map<number, FishDetail>()

const fishCount = parseInt(readline())
for (let i = 0; i < fishCount; i++) {
  const [fishId, color, type] = readline().split(' ').map(Number)
  fishDetails.set(fishId, { color, type })
}

// game loop
while (true) {
  const myScans: number[] = []
  const foeScans: number[] = []
  const droneById = new Map<number, Drone>()
  const myDrones: Drone[] = []
  const foeDrones: Drone[] = []
  const visibleFish: Fish[] = []
  const myRadarBlips = new Map<number, RadarBlip[]>()

  const myScore = parseInt(readline())
  const foeScore = parseInt(readline())

  const myScanCount = parseInt(readline())
  for (let i = 0; i < myScanCount; i++) {
    const fishId = parseInt(readline())
    myScans.push(fishId)
  }

  const foeScanCount = parseInt(readline())
  for (let i = 0; i < foeScanCount; i++) {
    const fishId = parseInt(readline())
    foeScans.push(fishId)
  }

  const myDroneCount = parseInt(readline())
  for (let i = 0; i < myDroneCount; i++) {
    const [droneId, droneX, droneY, dead, battery] = readline().split(' ').map(Number)
    const pos = { x: droneX, y: droneY }
    const drone = { droneId, pos, dead, battery, scans: [] }
    droneById.set(droneId, drone)
    myDrones.push(drone)
    myRadarBlips.set(droneId, [])
  }

  const foeDroneCount = parseInt(readline())
  for (let i = 0; i < foeDroneCount; i++) {
    const [droneId, droneX, droneY, dead, battery] = readline().split(' ').map(Number)
    const pos = { x: droneX, y: droneY }
    const drone = { droneId, pos, dead, battery, scans: [] }
    droneById.set(droneId, drone)
    foeDrones.push(drone)
  }


  const droneScanCount = parseInt(readline())
  for (let i = 0; i < droneScanCount; i++) {
    const [droneId, fishId] = readline().split(' ').map(Number)
    droneById.get(droneId)!.scans.push(fishId)
  }

  const visibleFishCount = parseInt(readline())
  for (let i = 0; i < visibleFishCount; i++) {
    const [fishId, fishX, fishY, fishVx, fishVy] = readline().split(' ').map(Number)
    const pos = { x: fishX, y: fishY }
    const speed = { x: fishVx, y: fishVy }
    visibleFish.push({fishId, pos, speed, detail: fishDetails.get(fishId)! })
  }

  const myRadarBlipCount = parseInt(readline())
  for (let i = 0; i < myRadarBlipCount; i++) {
    const [_droneId, _fishId, dir] = readline().split(' ')
    const droneId = parseInt(_droneId)
    const fishId = parseInt(_fishId)
    myRadarBlips.get(droneId)!.push({ fishId, dir })
  }

  for (const drone of myDrones) {
    const x = drone.pos.x
    const y = drone.pos.y
    // TODO: Implement logic on where to move here
    const targetX = 5000
    const targetY = 5000
    const light = 1

    console.log(`MOVE ${targetX} ${targetY} ${light}`)
  }
}

File: starterAIs/SS_Starter.go
package main

import (
	"fmt"
)

type Vector struct {
	X int
	Y int
}

type FishDetail struct {
	Color int
	Type  int
}

type Fish struct {
	FishID int
	Pos    Vector
	Speed  Vector
	Detail FishDetail
}

type RadarBlip struct {
	FishID int
	Dir    string
}

type Drone struct {
	DroneID int
	Pos     Vector
	Dead    bool
	Battery int
	Scans   []int
}

func main() {
	fishDetails := make(map[int]FishDetail)

	var fishCount int
	fmt.Scan(&fishCount)

	for i := 0; i < fishCount; i++ {
		var fishID, color, _type int
		fmt.Scan(&fishID, &color, &_type)
		fishDetails[fishID] = FishDetail{Color: color, Type: _type}
	}

	// Game loop
	for {
		var myScans, foeScans []int
		droneByID := make(map[int]Drone)
		var myDrones, foeDrones []Drone
		var visibleFish []Fish
		myRadarBlips := make(map[int][]RadarBlip)

		var myScore, foeScore int
		fmt.Scan(&myScore, &foeScore)

		var myScanCount int
		fmt.Scan(&myScanCount)
		for i := 0; i < myScanCount; i++ {
			var fishID int
			fmt.Scan(&fishID)
			myScans = append(myScans, fishID)
		}

		var foeScanCount int
		fmt.Scan(&foeScanCount)
		for i := 0; i < foeScanCount; i++ {
			var fishID int
			fmt.Scan(&fishID)
			foeScans = append(foeScans, fishID)
		}

		var myDroneCount int
		fmt.Scan(&myDroneCount)
		for i := 0; i < myDroneCount; i++ {
			var droneID, droneX, droneY, dead, battery int
			fmt.Scan(&droneID, &droneX, &droneY, &dead, &battery)
			pos := Vector{X: droneX, Y: droneY}
			drone := Drone{DroneID: droneID, Pos: pos, Dead: dead == 1, Battery: battery, Scans: []int{}}
			droneByID[droneID] = drone
			myDrones = append(myDrones, drone)
			myRadarBlips[droneID] = []RadarBlip{}
		}

		var foeDroneCount int
		fmt.Scan(&foeDroneCount)
		for i := 0; i < foeDroneCount; i++ {
			var droneID, droneX, droneY, dead, battery int
			fmt.Scan(&droneID, &droneX, &droneY, &dead, &battery)
			pos := Vector{X: droneX, Y: droneY}
			drone := Drone{DroneID: droneID, Pos: pos, Dead: dead == 1, Battery: battery, Scans: []int{}}
			droneByID[droneID] = drone
			foeDrones = append(foeDrones, drone)
		}

		var droneScanCount int
		fmt.Scan(&droneScanCount)
		for i := 0; i < droneScanCount; i++ {
			var droneID, fishID int
			fmt.Scan(&droneID, &fishID)
			drone := droneByID[droneID]
			drone.Scans = append(drone.Scans, fishID)
			droneByID[droneID] = drone
		}

		var visibleFishCount int
		fmt.Scan(&visibleFishCount)
		for i := 0; i < visibleFishCount; i++ {
			var fishID, fishX, fishY, fishVX, fishVY int
			fmt.Scan(&fishID, &fishX, &fishY, &fishVX, &fishVY)
			pos := Vector{X: fishX, Y: fishY}
			speed := Vector{X: fishVX, Y: fishVY}
			visibleFish = append(visibleFish, Fish{FishID: fishID, Pos: pos, Speed: speed, Detail: fishDetails[fishID]})
		}

		var myRadarBlipCount int
		fmt.Scan(&myRadarBlipCount)
		for i := 0; i < myRadarBlipCount; i++ {
			var droneID, fishID int
			var dir string
			fmt.Scan(&droneID, &fishID, &dir)
			myRadarBlips[droneID] = append(myRadarBlips[droneID], RadarBlip{FishID: fishID, Dir: dir})
		}

		for _, drone := range myDrones {
			x := drone.Pos.X
			y := drone.Pos.Y
			// TODO: Implement logic on where to move here
			targetX := 5000 + x - x
			targetY := 5000 + y - y
			light := 1

			fmt.Printf("MOVE %d %d %d\n", targetX, targetY, light)
		}
	}
}
